Client/Frontend/Browser/Playlist/Cache/PlaylistCacheLoader.swift

```diff
@@ -0,0 +1,277 @@
+        tab.addContentScript(ContextMenuHelper(tab: tab), name: ContextMenuHelper.name())
+        tab.addContentScript(ErrorPageHelper(), name: ErrorPageHelper.name(), sandboxed: false)
+        tab.addContentScript(SessionRestoreHelper(tab: tab), name: SessionRestoreHelper.name())
+        tab.addContentScript(FindInPageHelper(tab: tab), name: FindInPageHelper.name(), sandboxed: false)
+        tab.addContentScript(NoImageModeHelper(tab: tab), name: NoImageModeHelper.name())
+        tab.addContentScript(PrintHelper(tab: tab), name: PrintHelper.name())
+        tab.addContentScript(CustomSearchHelper(tab: tab), name: CustomSearchHelper.name())
+        tab.addContentScript(LocalRequestHelper(), name: LocalRequestHelper.name(), sandboxed: false)
+        tab.contentBlocker.setupTabTrackingProtection()
+        tab.addContentScript(tab.contentBlocker, name: ContentBlockerHelper.name(), sandboxed: false)
+        tab.addContentScript(FocusHelper(tab: tab), name: FocusHelper.name())
+        tab.addContentScript(FingerprintingProtection(tab: tab), name: FingerprintingProtection.name(), sandboxed: false)
+        tab.addContentScript(BraveGetUA(tab: tab), name: BraveGetUA.name())
+        if YubiKitDeviceCapabilities.supportsMFIAccessoryKey {
+            tab.addContentScript(U2FExtensions(tab: tab), name: U2FExtensions.name(), sandboxed: false)
+        }
+        tab.addContentScript(ResourceDownloadManager(tab: tab), name: ResourceDownloadManager.name(), sandboxed: false)
+        tab.addContentScript(WindowRenderHelperScript(tab: tab), name: WindowRenderHelperScript.name(), sandboxed: false)
+        tab.addContentScript(PlaylistHelper(tab: tab), name: PlaylistHelper.name(), sandboxed: false)
+        tab.addContentScript(PlaylistWebLoaderContentHelper(self), name: PlaylistWebLoaderContentHelper.name(), sandboxed: false)
+        
+        let script: WKUserScript? = {
+            guard let path = Bundle.main.path(forResource: "PlaylistDetector", ofType: "js"), let source = try? String(contentsOfFile: path) else {
+                return nil
+            }
+            
+            var alteredSource = source
+            let token = UserScriptManager.securityToken.uuidString.replacingOccurrences(of: "-", with: "", options: .literal)
+            alteredSource = alteredSource.replacingOccurrences(of: "$<videosSupportFullscreen>", with: "VSF\(token)", options: .literal)
+            
+            return WKUserScript(source: alteredSource, injectionTime: .atDocumentStart, forMainFrameOnly: false)
+        }()
+        
+        if let script = script {
+            tab.webView?.configuration.userContentController.addUserScript(script)
+        }
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func load(url: URL) {
+        guard let webView = tab.webView else { return }
+        webView.frame = self.window?.bounds ?? .zero
```

Comment:
Do we even need to set the frame ?

Reply:
Yes we do. Otherwise webView won't load properly. Some websites check the frame in order to determine if the webView loads as mobile or as desktop or a layout inbetween.